src/RobotViz.jl

```diff
@@ -149,25 +149,30 @@ function drawPoses(fg::G;
                    contour::Bool=true, levels::Int=1,
                    regexPoses=r"x"  ) where G <: AbstractDFG
     #
+    @info "drawPoses always sets orientation to max, regardless of meanmax setting.  TODO modefit."
     #Gadfly.set_default_plot_size(20cm, 30cm)
-    Xpp = Float64[]; Ypp=Float64[]; Thpp=Float64[]; LBLS=String[];
-    if meanmax == :mean
-      Xpp,Ypp, Thpp, LBLS = get2DPoseMeans(fg, from=from, to=to)
-    elseif meanmax == :max
-      Xpp,Ypp, Thpp, LBLS = get2DPoseMax(fg, from=from, to=to)
-    end
+    # Xpp = Float64[]; Ypp=Float64[]; Thpp=Float64[]; LBLS=String[];
+    uXpp,uYpp, uThpp, uLBLS = get2DPoseMeans(fg, from=from, to=to)
+    xXpp,xYpp, xThpp, xLBLS = get2DPoseMax(fg, from=from, to=to)
+        # if meanmax == :mean
+        # elseif meanmax == :max
+        # end
+    Xpp  = meanmax == :mean ? uXpp : xXpp
+    Ypp  = meanmax == :mean ? uYpp : xYpp
+    Thpp = xThpp # always use max -- should be modefit
+    LBLS = meanmax == :mean ? uLBLS : xLBLS
```

Comment:
Do you expect the labels do be different for the 2? If so it can be a problem. Otherwise it looks good.

Reply:
Ok, I see its sorted, then it should be good.

Reply:
oh yeah, this is hacky if you ask for `:something` you'll still just get max :-P but not a time critical thing and hoping community can help a little with expanding these features over time.